static/resources/harness.js

```diff
@@ -84,182 +102,74 @@
-
-                thisCompiled = 'CSS.supports(\'' +
-                    property + '\', \'inherit\');';
-                value = CSS.supports(property, 'inherit');
-              } else {
-                value = null;
-                result.message = 'Browser doesn\'t support CSS API';
-                break;
-              }
-            } else {
-              if (prefix) {
-                property = prefix + property.charAt(0).toUpperCase() +
-                           property.slice(1);
-              }
-
-              if (stringStartsWith(property, 'Symbol.')) {
-                thisCompiled = property+' in '+parentPrefix+subtest.scope;
-                value = eval(thisCompiled);
-              } else {
-                thisCompiled = '"'+property+'" in '+parentPrefix+subtest.scope;
-                value = eval(thisCompiled);
-              }
-
-              if (!compiled) {
-                // Set to first compiled statement in case support is false
-                compiled = thisCompiled;
-              }
-            }
-
-            result.result = value;
-            if (value === true) {
-              compiled = thisCompiled;
-
-              if (subtest.scope === 'CSS.supports') {
-                if (prefix) {
-                  parentPrefix = '-' + prefix + '-';
-                } else {
-                  parentPrefix = '';
-                }
-              } else {
-                parentPrefix = prefix;
-              }
-              break;
-            }
-          }
-
-          compiledCode.push(compiled);
+          result.result = null;
+          result.message = 'returned ' + stringify(value);
         }
+      } catch (err) {
+        result.result = null;
+        result.message = 'threw ' + stringify(err);
+      }
 
-        if (result.result === false) {
-          break;
-          // Tests are written in hierarchy order, so if the parent (first
-          // test) is unsupported, so is the child (next test)
+      if (result.result !== false) {
+        result.info.code = test.code;
+        if (test.prefix) {
+          result.info.prefix = test.prefix;
         }
-
-        result.prefix = parentPrefix;
+        break;
       }
-    } catch (err) {
-      result.result = null;
-      result.message = 'threw ' + stringify(err);
     }
 
     if (data.info !== undefined) {
       result.info = Object.assign({}, result.info, data.info);
     }
 
-    result.info.code = compiledCode.join(' && ');
+    if (result.result === false) {
+      result.info.code = data.tests[0].code;
+    }
     result.info.scope = data.scope;
 
     return result;
   }
 
-  function runCSS(callback) {
-    var results = [];
-
+  function runCSS(callback, results) {
     var length = pending.length;
     for (var i = 0; i < length; i++) {
-      updateStatus('Testing ' + pending[i].name);
-      results.push(test(pending[i]));
+      if (pending[i].scope == 'CSS') {
+        updateStatus('Testing ' + pending[i].name);
```

Comment:
This isn't new, but I'm nervous about doing really anything at all while the tests are running. Let's come back to this when moving more of the code generation to the server side.

Reply:
The `updateStatus()` isn't really working due to how much is going on in the JavaScript already (the DOM isn't updating), so these can be taken out safely.